crates/verify/src/sourcify.rs

```diff
@@ -25,16 +29,29 @@ impl VerificationProvider for SourcifyVerificationProvider {
         args: VerifyArgs,
         context: VerificationContext,
     ) -> Result<()> {
-        let _ = self.prepare_request(&args, &context)?;
+        let _ = self.prepare_verify_request(&args, &context).await?;
         Ok(())
     }
 
     async fn verify(&mut self, args: VerifyArgs, context: VerificationContext) -> Result<()> {
-        let body = self.prepare_request(&args, &context)?;
+        let body = self.prepare_verify_request(&args, &context).await?;
+        let chain_id = args.etherscan.chain.unwrap_or_default().id();
+
+        if !args.skip_is_verified_check && self.is_contract_verified(&args).await? {
+            sh_println!(
+                "\nContract [{}] {:?} is already verified. Skipping verification.",
+                context.target_name,
+                args.address.to_string()
+            )?;
+
+            return Ok(());
+        }
 
         trace!("submitting verification request {:?}", body);
 
         let client = reqwest::Client::new();
+        let base_url = args.verifier.verifier_url.as_deref().unwrap_or(SOURCIFY_URL);
+        let url = format!("{}v2/verify/{}/{}", base_url, chain_id, args.address);
```

Comment:
this repeats several times, let's have a helper / unified way to compose URL. Also, shouldn't `base_url` always be `SOURCIFY_URL` here, or we want to be flexible to change from `https://sourcify.dev/server/`?